### ISSUE TITLE
Rerunning GUI Error (Windows)

### ISSUE THREAD

**Issue by loganbeal** [BYU-PRISM/GEKKO#21]
The GUI can't be run twice from the same console on Windows 10. I ran it once successfully, stopped the process with Ctrl+C in the console, ran the script again and got this error:

 ```
 File "C:/Users/beall/Downloads/control3_fishing.py", line 34, in <module>
    m.GUI()

  File "C:\Anaconda2\lib\site-packages\gekko\gekko.py", line 581, in GUI
    gui.display()

  File "C:\Anaconda2\lib\site-packages\gekko\gk_gui.py", line 103, in display
    self.set_endpoints()

  File "C:\Anaconda2\lib\site-packages\gekko\gk_gui.py", line 72, in set_endpoints
    @app.route('/get_data')

  File "C:\Anaconda2\lib\site-packages\flask\app.py", line 1080, in decorator
    self.add_url_rule(rule, endpoint, f, **options)

  File "C:\Anaconda2\lib\site-packages\flask\app.py", line 64, in wrapper_func
    return f(self, *args, **kwargs)

  File "C:\Anaconda2\lib\site-packages\flask\app.py", line 1051, in add_url_rule
    'existing endpoint function: %s' % endpoint)

AssertionError: View function mapping is overwriting an existing endpoint function: get_data
```
It worked again in a new console.

**Comment by dhill2522** [comment]
This is also the case when the GUI is launched from Jupyter notebook
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-2-edc80526327d> in <module>()
     39 m.solve()
     40 
---> 41 m.GUI()

~/virenvs/gekkoDebug1/lib/python3.6/site-packages/gekko/gekko.py in GUI(self, **kwargs)
    579     def GUI(self, **kwargs):
    580         gui = GK_GUI(self.path,**kwargs)
--> 581         gui.display()

~/virenvs/gekkoDebug1/lib/python3.6/site-packages/gekko/gk_gui.py in display(self)
    101     def display(self):
    102         """Finds the appropriate port starts the api and opens the webbrowser"""
--> 103         self.set_endpoints()
    104         sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    105         port = 8050

~/virenvs/gekkoDebug1/lib/python3.6/site-packages/gekko/gk_gui.py in set_endpoints(self)
     70     def set_endpoints(self):
     71         """Sets the flask API endpoints"""
---> 72         @app.route('/get_data')
     73         def get_data():
     74             resp = jsonify(self.vars_dict)

~/virenvs/gekkoDebug1/lib/python3.6/site-packages/flask/app.py in decorator(f)
   1078         def decorator(f):
   1079             endpoint = options.pop('endpoint', None)
-> 1080             self.add_url_rule(rule, endpoint, f, **options)
   1081             return f
   1082         return decorator

~/virenvs/gekkoDebug1/lib/python3.6/site-packages/flask/app.py in wrapper_func(self, *args, **kwargs)
     62                 'database models and everything related at a central place '
     63                 'before the application starts serving requests.')
---> 64         return f(self, *args, **kwargs)
     65     return update_wrapper(wrapper_func, f)
     66 

~/virenvs/gekkoDebug1/lib/python3.6/site-packages/flask/app.py in add_url_rule(self, rule, endpoint, view_func, **options)
   1049             if old_func is not None and old_func != view_func:
   1050                 raise AssertionError('View function mapping is overwriting an '
-> 1051                                      'existing endpoint function: %s' % endpoint)
   1052             self.view_functions[endpoint] = view_func
   1053 

AssertionError: View function mapping is overwriting an existing endpoint function: get_data
```

**Comment by dhill2522** [comment]
This is also a problem in Spyder3.
```
Traceback (most recent call last):

  File "<ipython-input-2-bc07540a241f>", line 1, in <module>
    runfile('/home/dhill25/.config/spyder-py3/temp.py', wdir='/home/dhill25/.config/spyder-py3')

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/spyder/utils/site/sitecustomize.py", line 705, in runfile
    execfile(filename, namespace)

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/spyder/utils/site/sitecustomize.py", line 102, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "/home/dhill25/.config/spyder-py3/temp.py", line 41, in <module>
    m.GUI()

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/gekko/gekko.py", line 581, in GUI
    gui.display()

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/gekko/gk_gui.py", line 103, in display
    self.set_endpoints()

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/gekko/gk_gui.py", line 72, in set_endpoints
    @app.route('/get_data')

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/flask/app.py", line 1080, in decorator
    self.add_url_rule(rule, endpoint, f, **options)

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/flask/app.py", line 64, in wrapper_func
    return f(self, *args, **kwargs)

  File "/home/dhill25/virenvs/gekkoDebug1/lib/python3.6/site-packages/flask/app.py", line 1051, in add_url_rule
    'existing endpoint function: %s' % endpoint)

AssertionError: View function mapping is overwriting an existing endpoint function: get_data
```

**Comment by dhill2522** [comment]
This issue should be resolved by commit bbaeb25. Any interactive console should now be able to run the GUI multiple times without trouble.
